xocto/storage/s3_select.py

```diff
@@ -58,12 +58,9 @@ def to_dict(self) -> dict[str, Any]:
         temp_dict = {k: v for k, v in dataclasses.asdict(self).items() if v}
 
         for k, v in temp_dict.items():
-            if (
-                isinstance(v, FileHeaderInfo)
-                or isinstance(v, QuoteFields)
-                or isinstance(v, CompressionType)
-            ):
+            if type(v) in [FileHeaderInfo, QuoteFields, CompressionType]:
```

Comment:
I don't think this is the best way because `type` doesn't check for `inheritance`.  This link has a good [answer](https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance)